pkg/kubectl/cmd/describe.go

```diff
@@ -59,7 +59,7 @@ $ kubectl describe po -l name=myLabel`,
 
 func RunDescribe(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string) error {
 	selector := cmdutil.GetFlagString(cmd, "selector")
-	cmdNamespace, err := f.DefaultNamespace()
+	cmdNamespace, _, err := f.DefaultNamespace()
```

Comment:
I didn't look at enough of the context. 

Filed an issue for the other commands to accept -f.